### PR TITLE
8346046: Enable copyright header format check

### PR DESCRIPTION
**Pull request by zhaosongzs**
As requested in [SKARA-1357](https://bugs.openjdk.org/browse/SKARA-1357), now skara bot is able to check the copyright header format.
Now I am going to update the jcheck configuration to configure copyright check as an error.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8346046](https://bugs.openjdk.org/browse/JDK-8346046): Enable copyright header format check (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22721/head:pull/22721` \
`$ git checkout pull/22721`

Update a local copy of the PR: \
`$ git checkout pull/22721` \
`$ git pull https://git.openjdk.org/jdk.git pull/22721/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22721`

View PR using the GUI difftool: \
`$ git pr show -t 22721`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22721.diff">https://git.openjdk.org/jdk/pull/22721.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22721#issuecomment-2540230134)
</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
I think this looks ok, but would like input from at least another reviewer.

/reviewers 2

File: .jcheck/conf [hunk 1]
```diff
@@ -34,3 +34,8 @@ pattern=^([124-8][0-9]{6}): (\S.*)$
 
 [checks "problemlists"]
 dirs=test/jdk|test/langtools|test/lib-test|test/hotspot/jtreg|test/jaxp
+
+[checks "copyright"]
+files=^(?!LICENSE|license\.txt|.*\.bin|.*\.gif|.*\.jpg|.*\.png|.*\.patch|.*\.wav|.*\.class|.*-header|.*\.jar|).*
```
Comment (dholmes-ora) [inline .jcheck/conf:39]: IIUC there are other special files that could be excluded here: .icon, .tiff, .dat

But, again IIUC, this is just an optimization as the check would be "if the file has what appears to be an Oracle copyright, then validate the format". So binary files would not have it and so would not be validated.

**Review by erikj79 — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]

**Comment by zhaosongzs** [comment]
Thank you all for the reviews!
/integrate

**Pull request closed without merging**